caffe2/core/storage.h

```diff
@@ -32,6 +35,35 @@ class StorageImpl {
   explicit StorageImpl(DeviceType device_type) : device_type_(device_type) {}
   StorageImpl(DeviceType device_type, TypeMeta data_type)
       : data_type_(data_type), device_type_(device_type) {}
+  template <typename Deleter = MemoryDeleter>
+  StorageImpl(
+      DeviceType device_type,
+      TypeMeta data_type,
+      void* src,
+      TIndex size,
+      size_t capacity,
+      Deleter d = nullptr)
+      : data_type_(data_type), device_type_(device_type) {
+    CAFFE_ENFORCE_WITH_CALLER(
+        data_type_.id() != TypeIdentifier::uninitialized(),
+        "To create storage with a raw external pointer you need to have meta "
+        "already initialized.");
```

Comment:
I mentioned this previously, but "meta already initialized" is misleading for a constructor; you want to refer to the meta that is passed in.